docs/components.md

```diff
@@ -101,6 +101,109 @@ To learn more about lifecycle methods, [see the lifecycle methods page](lifecycl
 
 ---
 
+### Alternate component syntaxes
+
+#### ES6 classes
+
+Components can also be written using ES6 class syntax:
+
+```javascript
+class ES6ClassComponent {
+	constructor(vnode) {
+		// vnode.state is undefined at this point
```

Comment:
Could you mention that `vnode.state` will be set to the resulting instance *after* construction? It's worth noting that for consumers who aren't just sending it back into Mithril.